tokio-threadpool/src/builder.rs

```diff
@@ -267,12 +268,24 @@ impl Builder {
     /// # }
     /// ```
     pub fn build(&self) -> ThreadPool {
+        self.build_with_park(|| ::park::DefaultPark::new())
+    }
+
+    /// DOX
+    pub fn build_with_park<F, P>(&self, mut f: F) -> ThreadPool
```

Comment:
Rather then creating more `build_with_X` functions, should `ThreadPoolBuilder` be a thing?

Reply:
Well, this is on `Builder` :P. This could probably be moved to a builder fn instead of `build_with`. The initial impl (not in the PR) did not permit this.

Reply:
Wow completly missed it was on the `Builder` struct. But yeah a `fn with_park<F, P>(&mut self, val: F) -> &mut Self` would be better